programs/survival/compass.sc

```diff
@@ -0,0 +1,604 @@
+// If you want your script to work with the persist flag, signal the relevant scarpet event
+// For example: signal_event('player_places_block', player, player, l('acacia_door', 16, null), 'mainhand', block(0, 0, 0));
+
+__config()->{
+    'scope' -> 'global',
+    'commands' -> {
+        'giveCompass' -> _() -> giveCompass(player()),
+        'track <player>' -> ['track'],
+        'spawnWith item <item> <amt>' -> ['spawnWithItem', 'âny', false],
+        'spawnWith item <item> <amt> <player>' -> ['spawnWithItem', false],
+        'spawnWith item <item> <amt> persist' -> ['spawnWithItem', 'âny', true],
+        'spawnWith item <item> <amt> <player> persist' -> ['spawnWithItem', true],
+        'spawnWith effect <effect> <duration> <amplifier>' -> ['spawnWithEffect', 'âny'],
+        'spawnWith effect <effect> <duration> <amplifier> <player>' -> ['spawnWithEffect'],
+        'dontSpawnWith item <item>' -> ['dontSpawnWithItem', 'âny'],
+        'dontSpawnWith item <item> <player>' -> ['dontSpawnWithItem'],
+        'dontSpawnWith effect <effect>' -> ['dontSpawnWithEffect', 'âny'],
+        'dontSpawnWith effect <effect> <player>' -> ['dontSpawnWithEffect'],
+        'timeToTrack <time>' -> ['timeToTrack'],
+        'timeToTrack onUse' -> ['timeToTrack', 'onUse'],
+        'logOnTrack <enabled>' -> ['logOnTrack'],
+        'maxCompasses <amt>' -> ['maxCompasses'],
+        'maxCompasses infinity' -> ['maxCompasses', 'infinity'],
+        'noPortalBreaking <enabled>' -> ['noPortalBreaking'],
+        'maxNetherTravel <dist>' -> ['maxNetherTravel'],
+        'timeToTrack' -> _() -> print(player(), format('l How often the compasses update player positions in seconds: ', 'lb '+(global_timeToTrack/20))),
+        'track' -> _() -> print(player(), format('l Make the compass you\'re holding track a player')),
+        'spawnWith' -> _() -> print(player(), format('l Spawn with a particular item or effect when anyone or a specified player dies')),
+        'dontSpawnWith' -> _() -> print(player(), format('l Stop spawning with a particular item or effect')),
+        'logOnTrack' -> _() -> print(player(), format('l Send a message when someone tracks you: ', 'lb '+if(global_logOnTrack, 'true', 'false'))),
+        'maxCompasses' -> _() -> print(player(), format('l The maximum amount of compasses one can give themselves: ', 'lb '+global_maxCompasses)),
+        'noPortalBreaking' -> _() -> print(player(), format('l Whether to allow breaking portals: ', 'lb '+if(global_noPortalBreaking, 'true', 'false'))),
+        'maxNetherTravel' -> _() -> print(player(), format('l The maximum distance players can travel in the overworld thorugh the nether: ', 'lb '+(global_maxNetherTravel*8))),
+    },
+    'arguments' -> {
+        'player' -> {'type' -> 'string', 'suggester' -> _(args) -> entity_selector('@a'), 'case_sensitive' -> false},
```

Comment:
You can replace `entity_selector('@a')` with `player('all')`. It is true that that may conflict with a player whose name is 'all', but 3-letter names and less are no longer a thing, and player('all') is hardcoded to work and capture all players in all dimensions.

Reply:
I did some testing on my machine, and found that `entity_selector('@a')` was consistently much faster than `player('all')`, however, they're both so fast that I'm not sure it really matters anyways

```
/script run for(range(1000000), _)                                    // around 100ms
/script run for(range(1000000), entity_selector('@a'))                // around 300ms
/script run for(range(1000000), player('all'))                        // around 450ms
```

I'd be curious to see what your results are, or if you think this is even a fair test

Reply:
better to do:
```
/script run profile_expr(_)
/script run profile_expr(entity_selector('@a'))
/script run profile_expr(player('all'))
```
cos that (i think) runs it better to test it to your system.
I'd check those, and then see.

Reply:
The `entity_selector('@a')` one was still faster for me